uni-pad/src/native_pad.rs

```diff
@@ -1,10 +1,46 @@
-// TODO
-pub fn gamepad_init() {}
+use gilrs::Gilrs;
+use gilrs::ev::{Axis, Button};
 
-pub fn gamepad_axis(_player_num: i32) -> (f32, f32) {
+static mut CTX: Option<Gilrs> = None;
+
+static BUTTON_TO_NECS: [Button; 4] = [Button::South, Button::East, Button::West, Button::North];
+
+pub fn gamepad_init() {
+    unsafe {
+        CTX = Gilrs::new().ok();
+        if let Some(ref ctx) = CTX {
+            for (id, gamepad) in ctx.gamepads() {
+                println!(
+                    "gamepad[{}] {} is {:?}",
+                    id,
+                    gamepad.name(),
+                    gamepad.power_info()
+                );
+            }
+        } else {
+            println!("warning : no gamepad support");
+        }
+    }
+}
+
+pub fn gamepad_axis(player_num: i32) -> (f32, f32) {
+    unsafe {
```

Comment:
Why all these methods are unsafe ?

Reply:
Because usage of a static mut (CTX). So far I kept the basic API used in the web part. But maybe we should refactor it into a struct so that we don't need a static variable.

Reply:
We could use [Once](https://doc.rust-lang.org/beta/std/sync/struct.Once.html) to initialized the gilrs instance  and use the add a `get()` function to access it as similar to [this example](https://doc.rust-lang.org/beta/std/sync/struct.Once.html#examples-1).